src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -148,7 +148,11 @@ public function checkRequestPath(Request $request, string $path)
      */
     public function generateUri(Request $request, string $path)
     {
-        if (str_starts_with($path, 'http') || !$path) {
+        $parsedUrl = parse_url($path);
```

Comment:
```suggestion
        $url = parse_url($path);
```

+ remove the net lines :)